### ISSUE TITLE
Consider iterating over the string in forward direction

### ISSUE THREAD

**Issue by victor-homyakov** [darkskyapp/string-hash#12]
> Differences include iterating over the string backwards (as that is faster in JavaScript)

This requires some proof, because for long strings backward iteration may break any data caching/prefetching, therefore access to string characters may be significantly slower.

https://jsperf.com/string-hash-speed forward direction is faster in Chrome

https://gist.github.com/victor-homyakov/bcb7d7911e4a388b1c810f8c3ce17bcf shows me that in Node 8.6.0 forward iteration is twice as fast.

**Comment by ironwallaby** [comment]
Yes, this is almost certainly correct now. JS has evolved tremendously since this library was put to paper (six years ago!), and I've been loathe to change it (since it's a breaking change and perhaps people rely on the hash values being consistent).

**Comment by dguo** [comment]
@ironwallaby would it make sense to do so by releasing the change with a new major version?

**Comment by ghost** [comment]
We won't be doing that from here, since this library is no longer maintained. Please fork and go ahead! (And, if you like, link us to the fork and we'll link to it from the README.)
